bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitChangeListener.java

```diff
@@ -221,25 +249,64 @@ private Optional<Item> getItemOptional(String name) {
     public void makeNewConfigurationRevision() {
         final int newRevision = accessoryRegistry.makeNewConfigurationRevision();
         lastAccessoryCount = accessoryRegistry.getAllAccessories().size();
-        logger.trace("Make new configuration revision. new revision number {}, number of accessories {}", newRevision,
-                lastAccessoryCount);
+        logger.info("Created {} HomeKit items in instance {}.", accessoryRegistry.getAllAccessories().size(), instance);
+        checkForDummyAccessories();
+        logger.trace("Making new configuration revision {}", newRevision);
         storage.put(REVISION_CONFIG, "" + newRevision);
-        storage.put(ACCESSORY_COUNT, "" + lastAccessoryCount);
+        storage.put(KNOWN_ACCESSORIES, knownAccessories);
+    }
+
+    public synchronized void pruneDummyAccessories() {
+        boolean removed = false;
+        for (HomekitAccessory accessory : accessoryRegistry.getAllAccessories().values()
```

Comment:
i would use a filter and for each here as well.
e.g.
`
accessoryRegistry.getAllAccessories().values().filter((a -> a instanceof DummyHomekitAccessory).forEach(accessory -> {
                try { ...
`

Reply:
right. missed that. thanks